akka-http-core/src/main/scala/akka/http/scaladsl/settings/Http2ServerSettings.scala

```diff
@@ -26,6 +29,24 @@ private[http] trait Http2CommonSettings {
   def logFrames: Boolean
   def maxConcurrentStreams: Int
   def outgoingControlFrameBufferSize: Int
+
+  def pingInterval: FiniteDuration
+  def pingTimeout: FiniteDuration
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[http] object Http2CommonSettings {
+  def validate(settings: Http2CommonSettings): Unit = {
+    import settings._
+    if (pingInterval > Duration.Zero && pingTimeout > Duration.Zero) {
+      require(
+        pingTimeout <= pingInterval && pingInterval.toMillis % pingTimeout.toMillis == 0,
+        s"ping-timeout must be less than and evenly divisible by the ping-interval ($pingInterval)")
```

Comment:
I guess that's not too much of a restriction? Right now I guess we envision settings like 10 seconds for interval and 1 second for timeout. Might become weird if you want to set the timeout to very specific values (but maybe it's not necessary to worry about that?).

The description seems to be backwards, though?

```suggestion
        s"ping-timeout must be less than and a divisor of the ping-interval ($pingInterval)")
```

Reply:
🤦